### PR TITLE
UI: Updated in settings

### PR DESCRIPTION
**Pull request by maxime-desroches**
Follows from #21036

- Adds the Last Update Time label in the developer settings.

- Clicking on Last Update Time sends a SIGHUP to updated.

- The description of the Version label is the release notes.

I had to modify the constructor of LabelControl in order to add a description to some labels.

Let me know if something is wrong/bad with this

### CODE REVIEW

File: selfdrive/ui/qt/widgets/controls.h [hunk 1]
```diff
@@ -42,7 +42,7 @@ class LabelControl : public AbstractControl {
   Q_OBJECT
 
 public:
-  LabelControl(const QString &title, const QString &text, const QString &desc = "", QWidget *parent = nullptr) : AbstractControl(title, desc, "", parent) {
+  LabelControl(const QString &title, const QString &text, const QString &desc, QWidget *parent = nullptr) : AbstractControl(title, desc, "", parent) {
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/widgets/controls.h:45]: why change the default?

File: selfdrive/ui/qt/offroad/settings.cc [hunk 2]
```diff
@@ -214,20 +214,39 @@ void DeveloperPanel::showEvent(QShowEvent *event) {
...
+  std::string lastUpdateTimeRaw = params.get("LastUpdateTime", false);
+  QString version = QString::fromStdString(brand + " v" + params.get("Version", false).substr(0, 14)).trimmed();
+  QString lastUpdateTime = QString::fromStdString(lastUpdateTimeRaw.substr(0,10) + " at " + lastUpdateTimeRaw.substr(12,7)).trimmed();
+
+  if (labels.size() < dev_params.size()) {
+    versionLbl = new LabelControl("Version", version, QString::fromStdString(params.get("ReleaseNotes", false)).trimmed());
+    layout()->addWidget(versionLbl);
+    layout()->addWidget(horizontal_line());
+
+    lastUpdateTimeLbl = new LabelControl("Last Update Time", lastUpdateTime, "The last time openpilot checked for an update");
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/offroad/settings.cc:232]: let's call this "Last Update Check"

File: selfdrive/ui/qt/offroad/settings.cc [hunk 3]
```diff
@@ -214,20 +214,39 @@ void DeveloperPanel::showEvent(QShowEvent *event) {
...
-    {"Version", brand + " v" + params.get("Version", false).substr(0, 14)},
     {"Git Branch", params.get("GitBranch", false)},
     {"Git Commit", params.get("GitCommit", false).substr(0, 10)},
     {"Panda Firmware", params.get("PandaFirmwareHex", false)},
     {"OS Version", Hardware::get_os_version()},
   };
 
+  std::string lastUpdateTimeRaw = params.get("LastUpdateTime", false);
+  QString version = QString::fromStdString(brand + " v" + params.get("Version", false).substr(0, 14)).trimmed();
+  QString lastUpdateTime = QString::fromStdString(lastUpdateTimeRaw.substr(0,10) + " at " + lastUpdateTimeRaw.substr(12,7)).trimmed();
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/offroad/settings.cc:225]: is there some nice Qt function to get us a nicely formatted time like "one day ago"?

File: selfdrive/ui/qt/widgets/controls.h (hunk 1 above)
Comment (maxime-desroches) [reply selfdrive/ui/qt/widgets/controls.h]: This is indeed unnecessary, I changed it back

File: selfdrive/ui/qt/offroad/settings.cc (hunk 3 above)
Comment (maxime-desroches) [reply selfdrive/ui/qt/offroad/settings.cc]: I wasn't able to find any Qt functions for this exact format ("one day ago"), but I used the QDateTime::daysTo() to make something similar

Comment (adeebshihadeh) [reply selfdrive/ui/qt/offroad/settings.cc]: let's make a function in `selfdrive/ui/qt/util.h` that returns the string given a QDateTime. it should output:

Comment (maxime-desroches) [reply selfdrive/ui/qt/offroad/settings.cc]: Done. I also added   x hours ago, when < 1 day

**Comment by adeebshihadeh** [comment]
Thanks for the PR!

**Pull request merged**